woodwork/utils.py

```diff
@@ -638,8 +639,26 @@ def _infer_datetime_format(dates, n=100):
     try:
         fmts = first_n.map(pd.core.tools.datetimes.guess_datetime_format)
         mode_fmt = fmts.mode().loc[0]  # select first most common format
-    except (TypeError, ValueError, IndexError, KeyError, NotImplementedError):
```

Comment:
@jeff-hernandez @bchen1116 There were a lot of general exceptions being caught here. Do we know why we needed so many? It seems that after my change, keeping these exceptions resulted in no code coverage for this line, so `IndexError`, `KeyError`, and `NotImplementedError` weren't being tested.